### ISSUE TITLE
Saving left 0 padded strings (e.g. "030") as integers cause saving to fail silently

### ISSUE THREAD

**Issue by endoscient** [datamapper/dm-core#228]
```ruby
class Foo
  include DataMapper::Resource  
  property :id, Serial
  property :number, Integer
end

f = Foo.new({ :number => "030" })
f.valid? # true
f.save # false
f.errors # {}
```

Using latest released version of everything, with sqlite adpator.

Solution. Either have it return a validation error, or parse the string to an integer.

**Comment by tpitale** [comment]
The pull request was closed. I assume it was merged into release-1.2, but it does not appear in master. I'm not sure what happened to the release-1.2 branch, it's also not in the v1.2.0 tag.

**Comment by FeeJai** [comment]
Hey. It wasn't merged because I did not provide any specs. I do not work with data mapper anymore. The bug still exists though and is severe
